### PR TITLE
[loader] Test for #12166

### PR DESCRIPTION
**Pull request by runtian-zhou**
### Description

Implemented a testsuite for #12166. We randomly created a module dependency graph and make sure we charge transitive dependencies properly for each module entry function.

### Test Plan

Test added

### CODE REVIEW

File: aptos-move/e2e-tests/src/loader/mod.rs [hunk 1]
```diff
@@ -295,6 +336,42 @@ impl DependencyGraph {
...
+        self.caculate_dependency_sizes();
+        for account_idx in accounts.iter() {
+            let txn = self.invoke_at(account_idx);
+            let (output, log) = executor.execute_transaction_with_gas_profiler(txn).unwrap();
+            assert_eq!(
+                output.status(),
+                &TransactionStatus::Keep(ExecutionStatus::Success)
+            );
+            let node = self.graph.node_weight(*account_idx).unwrap();
+            assert!(
```
Comment (vgao1996) [inline aptos-move/e2e-tests/src/loader/mod.rs:363]: Would also be helpful to check the number of dependencies.

File: aptos-move/e2e-testsuite/src/tests/loader.rs [hunk 2]
```diff
@@ -47,4 +47,19 @@ proptest! {
     ) {
         run_and_assert_universe(universe, additional_txns);
     }
+
+    #[test]
+    fn smaller_world_and_test_deps_charging(
```
Comment (vgao1996) [inline aptos-move/e2e-testsuite/src/tests/loader.rs:52]: We might also want to test module publishing. It's not urgent though so maybe just leave a TODO?

**Review by vgao1996 — APPROVED** [review]
LGTM. @runtian-zhou do you want me to cherry-pick your commit to my PR or let you land this separately?

**Comment by runtian-zhou** [comment]
> LGTM. @runtian-zhou do you want me to cherry-pick your commit to my PR or let you land this separately?

Feel free to cherry-pick.

**Pull request closed without merging**